invoice/src/main/java/org/killbill/billing/invoice/dao/DefaultInvoiceDao.java

```diff
@@ -767,8 +774,11 @@ public boolean apply(final InvoicePaymentModelDao input) {
                 // Notify the bus since the balance of the invoice changed
                 final UUID accountId = transactional.getAccountIdFromInvoicePaymentId(chargeBack.getId().toString(), context);
 
-                cbaDao.doCBAComplexityFromTransaction(payment.getInvoiceId(), invoicesTags, entitySqlDaoWrapperFactory, context);
-
+                final Set<UUID> modifiedInvoiceIds = new HashSet<>();
+                modifiedInvoiceIds.addAll(cbaDao.doCBAComplexityFromTransaction(ImmutableSet.of(payment.getInvoiceId()), invoicesTags, entitySqlDaoWrapperFactory, context));
```

Comment:
Nit: the pattern

1. create set to track ids
2. run CBA
3. notify bus

repeats itself a few times - might be possible to abstract a function?

Reply:
See cb5ae6d104cf